### ISSUE TITLE
Consider more space-efficient reply rendering

### ISSUE THREAD

**Issue by jtdaugherty** [matterhorn-chat/matterhorn#490]
@kquick and I discussed some ways to improve the space efficiency of the way we render reply indicators. One idea we came up with was the following.

If we have two adjacent messages in the same thread, only render the thread root message once, and indent the replies together. So instead of
```
user1: I like sandwiches
user2: this message is about something else
 ┌▸user1: I like sandwiches
user3: I disagree
 ┌▸user1: I like sandwiches
user4: sandwiches are the best
```
we could have
```
user1: I like sandwiches
user2: this message is about something else
 ┌▸user1: I like sandwiches
 │ user3: I disagree
 │ user4: sandwiches are the best
```
or
```
user1: I like sandwiches
user2: this message is about something else
[reply] user1: I like sandwiches
 │ user3: I disagree
 │ user4: sandwiches are the best
```
and if the replies are adjacent to the root post without an intervening unrelated message, as in the case above, then we might have
```
user1: I like sandwiches
 │ user3: I disagree
 │ user4: sandwiches are the best
```

**Comment by david-christiansen** [comment]
I'm personally a fan of
```
user1: I like sandwiches
user2: this message is about something else
[reply] user1: I like sandwiches
 │ user3: I disagree
 │ user4: sandwiches are the best
```
because it shows the hierarchical structure clearly and it has a bit of text to help figure out what it means, without filling a lot of space.

**Comment by jtdaugherty** [comment]
Thank you, @david-christiansen!

**Comment by jtdaugherty** [comment]
I'm working on this now on `feature/reply-coalescing`. It turns out that the `[reply]` bit is noisy enough to break readability, and it looks nicer as just
```
[10:01] user1: I like sandwiches
[10:02] user2: this message is about something else
        user1: I like sandwiches
[10:03] │ user3: I disagree
[10:04] │ user4: sandwiches are the best
```
where the `user1` message that appears the second time is rendered in grayscale, which helps it appear as just informative. It also helped to align the timestamps with the replies, not with the parent messages, as I did above.

**Comment by jtdaugherty** [comment]
(This also means that the way the replies are rendered is identical in each case - the case where the parent is adjacent and the case where it isn't and has to be re-drawn.)

**Comment by jtdaugherty** [comment]
The UI tweaks are now done in `feature/reply-coalescing`. Once @kquick is available we can do a review.

**Comment by jtdaugherty** [comment]
@kquick and I spotted a few issues that need to be addressed:
* The `New Messages` line's interaction with reply rendering needs to be changed so that when that line occurs within two otherwise adjacent thread posts, the second one's parent is not re-rendered.
* The "|" bar that appears to the left of replies does not extend to cover attachments and reactions.

**Comment by jtdaugherty** [comment]
Another issue that needs to be resolved: the logic for word-wrapping messages does not take the extra reply indentation into account, thus truncating two columns on the right side of the rendered message.

**Comment by jtdaugherty** [comment]
All three of the issues detailed above have now been addressed.

**Comment by jtdaugherty** [comment]
This is now released in 50200.3.0.
